R/zchunk_LA154.Transport_china.R

```diff
@@ -0,0 +1,196 @@
+        ungroup() ->
+        NBS_transportation_national
+
+      # Now the province shares can be calculated by dividing the province data by the national
+      L154.NBS_trn_Mtce_province %>%
+        left_join_error_no_match(NBS_transportation_national, by = c("EBProcess", "EBMaterial", "year")) %>%
+        mutate(value_share = value / value_national) %>% # Calculating province's share
+        # NAs were introduced where national values were 0. Replace NAs with zeros.
+        replace_na(list(value_share = 0)) %>%
+        select(province, EBProcess, EBMaterial, year, value_share) ->
+        L154.NBS_trn_share_province
+
+      # Match these percentages into a table of all transportation technologies that will be written out
+      L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh %>%
+        repeat_add_columns(tibble::tibble(province = gcamchina.PROVINCES)) ->
+      L154.pct_province_trn_m_sz_tech_F_Yh
+
+      # TODO: Tibet
+      #L154.pct_province_trn_m_sz_tech_F_Yh <- subset( L154.pct_province_trn_m_sz_tech_F_Yh, province != "XZ" )
+
+      L154.pct_province_trn_m_sz_tech_F_Yh %>%
+        filter(year %in% HISTORICAL_YEARS) %>%
+        mutate(fuel_sector = paste(EBProcess, EBMaterial)) %>%
+        left_join(L154.NBS_trn_share_province, by = c("province", "EBProcess", "EBMaterial", "year")) %>%
+        replace_na(list(value_share = 0)) %>%
+        mutate(value = NULL) ->
+        L154.pct_province_trn_m_sz_tech_F_Yh
+
+      #Now, the full CHINA tran UCD database can be apportioned to the provinces
+      # Creating the first of the three output tables
+      L154.pct_province_trn_m_sz_tech_F_Yh%>%
+        left_join_error_no_match(L154.in_EJ_CHINA_trn_m_sz_tech_F_Yh, by = c("year", "UCD_sector", "mode", "size.class", "UCD_technology", "UCD_fuel","fuel")) %>%
+        mutate(value = value * value_share) %>% # Allocating across the provinces
+        select(province, UCD_sector, mode, size.class, UCD_technology, UCD_fuel, fuel, year, value) ->
+        L154.in_EJ_province_trn_m_sz_tech_F
+
+      # As a final step, aggregate by fuel and name the sector
+      # This creates the second of three output tables
+      L154.in_EJ_province_trn_m_sz_tech_F %>%
+        group_by(province, fuel, year) %>%
+        summarise(value = sum(value)) %>%
+        ungroup() %>%
+        mutate(sector = "transportation") %>% # Adding a column named "sector" with "transportation" as the entries
+        select(province, sector, fuel, year, value) ->
+        L154.in_EJ_province_trn_F
+
+      # Apportion non-motorized energy consumption to provinces on the basis of population
+      # First we will create the province shares based on population
+      L100.Pop_thous_province %>%
+        complete( province, year = 1971:2100) %>%
+        group_by(province) %>%
+        mutate(pop = approx_fun(year, pop, rule = 2)) %>%
+        ungroup() %>%
+        group_by(year) %>%
+        summarise(value_national = sum(pop, na.rm =T)) ->
```

Comment:
again, no need to calculate the national total separately, you can simply do:
`group_by(year) %>%`
`mutate(share = pop / sum(pop, na.rm = T)) %>%`